azure-pipelines.yml

```diff
@@ -88,6 +88,7 @@ jobs:
         DISTRIB: 'conda-free-threaded'
         LOCK_FILE: './build_tools/azure/pylatest_free_threaded_linux-64_conda.lock'
         COVERAGE: 'false'
+        SKLEARN_FAULTHANDLER_TIMEOUT: '1800'  # 30 * 60 seconds
```

Comment:
The usual duration for the `pylatest_free_threaded` run is below 20 min, so 30 minutes should give a large enough margin to avoid false positive.